pkg/client/token_service.go

```diff
@@ -93,14 +93,17 @@ func (ts *tokenService) parseContent() (string, string, error) {
 	if strings.HasPrefix(content, "v2.public.") {
 		return "", "", errors.New("old token format. Please remove old token located in your default home dir")
 	}
-	databasel := make([]byte, 8)
-	copy(databasel, content[:8])
-	databaselUint64 := binary.BigEndian.Uint64(databasel)
-	databasename := make([]byte, int(databaselUint64))
-	copy(databasename, content[8:8+int(databaselUint64)])
-
-	token := make([]byte, len(content)-8-int(databaselUint64))
-	copy(token, content[8+int(databaselUint64):])
+	dbnamelen := make([]byte, 8)
+	copy(dbnamelen, content[:8])
+	dbnamelenUint64 := binary.BigEndian.Uint64(dbnamelen)
+	if int(dbnamelenUint64) > len(content) {
```

Comment:
yes, actually should not be greater than remaining content, 8 bytes were already consumed when reading the encoded length

Reply:
@jeroiraz https://play.golang.org/p/WT-Nin6UVBb
checking that dbnamelenUint64. is not bigger than the whole token avoids the panic

I do not think that casting to int is needed  if uint64(len(content)) is used https://play.golang.org/p/cdH3lMq5TC0

Reply:
I mean something like
lenb := make([]byte, 8)
copy(lenb, content)

dbNameLen := int(binary.BigEndian.Uint64(lenb))

 `if dbNameLen > len(content)-8` ...  // note: here dbNameLen should be less than whole content less 8 bytes used to encode the len of the dbName itself